COCO_DATA_FORMAT.md

```diff
@@ -110,6 +110,21 @@ Here is one example of the json file for the image regression task, where the "t
 }
 ```
 
+## Image retrieval
+
+Here is one example of the json file for the image retrieval task. The image retrieval json file is similar to image_caption and image text matching dataset formats in that it contains text associated to images. However, there are some important differences: 1) There is an optional category section which allows the definition of both a category name and a super category. 2) Unlike the other 2 dataset formats, the annotations contain a "query" field rather than "text" or "caption" to match the task at hand.
```

Comment:
It might be better to explain it in a layman terms, I annot really understand why an annotation has an image id and a category id and a query